test/channels.jl

```diff
@@ -39,6 +40,22 @@ end
     @test fetch(t) == "finished"
 end
 
+@testset "timed wait on Condition" begin
+    a = Condition()
+    @test_throws ArgumentError wait(a; timeout=0.0005)
+    @test_throws TimeoutError wait(a; timeout=0.1)
+    @spawn begin
+        sleep(0.01)
+        notify(a)
+    end
+    @test try
+        wait(a; timeout=2)
+        true
+    catch
+        false
+    end
```

Comment:
This is a racy test. I think you could make it not racy by using `@async`?

Reply:
Ah yeah, that `sleep` is too short and `@async` would be safer here.